src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -477,6 +484,55 @@ public AgentSettings LoadSettings()
             }
         }
 
+        private void ConfigureAutoLogonIfNeeded(CommandSettings command)
+        {
+            bool enableAutoLogon = command.GetEnableAutoLogon();
+            if(!enableAutoLogon)
+            {
+                return;
+            }
+
+            try
+            {
+                AssertAdminAccess(command);
+                
+                var iConfigManager = HostContext.GetService<IInteractiveSessionConfigurationManager>();
+                iConfigManager.Configure(command);
+
+                var processHelper = HostContext.GetService<IProcessInvoker>();
+                if(iConfigManager.RestartNeeded())
+                {
+                    _term.WriteLine(StringUtil.Loc("RestartMessage"));
+                    var shallRestart = command.GetRestartPermission();
+                    if(shallRestart)
```

Comment:
the agent is configured with 'interactivesession' capability. He'll have to manually call run.cmd to get it up and honoring jobs or will have to restart the machine.